### PR TITLE
chore: bump golang.org/x/crypto to v0.21.0

### LINKED ISSUE #126: security: Upgrade golang.org/x/crypto to remediate CVEs
Chisel is being reported as containing High Risk Vulnerabilities do to the version of golang.org/x/crypto used in release v0.9.0.

image

golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
Chisel v0.9.0

Can this please be updated and published as a new release of Chisel?

### PR DESCRIPTION
**Pull request by rebornplusplus**
This PR bumps the version of [golang.org/x/crypto](golang.org/x/crypto) to v0.21.0.

Changes reflect the following operation:
```
go get -u golang.org/x/crypto
```

Resolves #126

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -43,6 +43,8 @@ go.starlark.net v0.0.0-20220328144851-d1966c6b9fcd h1:Uo/x0Ir5vQJ+683GXB9Ug+4fcj
 go.starlark.net v0.0.0-20220328144851-d1966c6b9fcd/go.mod h1:t3mmBBPzAVvK0L0n1drDmrQsJ8FoIx4INCqVMTr/Zo0=
 golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2 h1:VklqNMn3ovrHsnt90PveolxSbWFaJdECFbxSq0Mqo2M=
 golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2/go.mod h1:djNgcEr1/C05ACkg1iLfiJU5Ep61QUkGW8qpdssI0+w=
```
Comment (letFunny) [inline go.sum:45]: Should we also run `go mod tidy` to see if one of these entries is redundant?

File: .golangci.yaml [hunk 2]
```diff
@@ -51,5 +51,9 @@ issues:
       text: "globalLogger.*unused"
       linters:
         - unused
+    - path: "^.*.go$"
+      text: "golang.org/x/crypto/openpgp.*deprecated"
+      linters:
+        - staticcheck
```
Comment (letFunny) [inline .golangci.yaml:57]: Why do we need this?

Comment (rebornplusplus) [reply .golangci.yaml:57]: To suppress the `staticcheck` linting error:

```
internal/archive/testarchive/testarchive.go:11:2: SA1019: "golang.org/x/crypto/openpgp/clearsign" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/clearsign"
	^
internal/archive/testarchive/testarchive.go:12:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/archive/archive.go:11:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/testutil/pgpkeys.go:6:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/pgputil/openpgp.go:8:2: SA1019: "golang.org/x/crypto/openpgp/armor" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/armor"
	^
internal/pgputil/openpgp.go:9:2: SA1019: "golang.org/x/crypto/openpgp/clearsign" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/clearsign"
	^
internal/pgputil/openpgp.go:10:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/setup/setup_test.go:8:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/pgputil/openpgp_test.go:4:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/setup/setup.go:12:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
internal/archive/archive_test.go:4:2: SA1019: "golang.org/x/crypto/openpgp/packet" is deprecated: this package is unmaintained except for security fixes. New applications should consider a more focused, modern alternative to OpenPGP for their specific task. If you are required to interoperate with OpenPGP systems and need a maintained package, consider a community fork. See https://golang.org/issue/44226. (staticcheck)
	"golang.org/x/crypto/openpgp/packet"
	^
```

Comment (letFunny) [reply .golangci.yaml:57]: Thanks! Only one request, can we change the regexp to be as specific as possible? Something like:
```
\"golang.org/x/crypto/openpgp/\w+\" is deprecated
```

Comment (rebornplusplus) [reply .golangci.yaml]: Updated, thanks! https://github.com/canonical/chisel/pull/130/commits/4dbee41d64c46077528f26482c7a7333b2729b09

**Review by letFunny — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -9,14 +9,14 @@ require (
 	github.com/klauspost/compress v1.15.4
 	github.com/ulikunitz/xz v0.5.10
 	go.starlark.net v0.0.0-20220328144851-d1966c6b9fcd
-	golang.org/x/crypto v0.0.0-20190308221718-c2843e01d9a2
-	golang.org/x/term v0.12.0
+	golang.org/x/crypto v0.21.0
+	golang.org/x/term v0.18.0
```
Comment (dolmen) [inline go.mod:13]: This upgrade is out of scope of this PR.

**Review by dolmen — CHANGES_REQUESTED** [review]

File: go.mod (hunk 3 above)
Comment (rebornplusplus) [reply go.mod:13]: `golang.org/x/term` is required for `golang.org/x/crypto` actually: https://cs.opensource.google/go/x/crypto/+/master:go.mod.

```go
module golang.org/x/crypto

go 1.18

require (
	golang.org/x/net v0.21.0 // tagx:ignore
	golang.org/x/sys v0.19.0
	golang.org/x/term v0.19.0
)

require golang.org/x/text v0.14.0 // indirect
```

**Review by niemeyer — APPROVED** [review]
Thanks, Rafid.
